### PR TITLE
signed macos bin 

### PR DESCRIPTION
**Pull request by kacurez**
fixes https://keboola.atlassian.net/browse/SRE-2617
To ze to funguje som si overil tak ze som si uploadol `signed` artifacty z predchadzajuceho buildu, pustil lokalne a fungovalo, tj rozbaleny `kbc` binary nehlasi `unsigned binary error` vid build https://github.com/keboola/keboola-as-code/actions/runs/1198430806

image
image

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -16,13 +17,31 @@ builds:
...
+      post: '{{ if index .Env "TARGET_PATH"  }}cp "{{ .Path }}" "{{ .Env.TARGET_PATH }}"{{ else }}true{{ end }}'
 
 changelog:
   skip: true
 
 archives:
   - files:
       - none*
+    builds:
+      - kbc
```
Comment (kacurez) [inline .goreleaser.yml:44]: zipuje iba non darwin archivy

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -134,3 +137,57 @@ jobs:
...
+        with:
+          name: darwin-binaries
+          path: release/
+      - name: Prepare files
+        run: |
+          mkdir release/signed
+          mkdir release/darwin-amd64
+          cp release/*amd64/kbc release/darwin-amd64
+          mkdir release/darwin-arm64
+          cp release/*arm64/kbc release/darwin-arm64
```
Comment (kacurez) [inline .github/workflows/push.yml:161]: kopirujem si do zvlastnych zloziek `release/darwin-arm64` resp `amd64` pretoze v gon.json configu musi byt pevna cesta, po release je dynamicka(obsahuje verziu) a to do gon.json nemozem dat

File: .github/workflows/push.yml [hunk 3]
```diff
@@ -134,3 +137,57 @@ jobs:
...
+        env:
+          AC_USERNAME: ${{ secrets.AC_USERNAME }}
+          AC_PASSWORD: ${{ secrets.AC_PASSWORD }}
+        run: |
+          gon -log-level=debug -log-json ./gon-amd64.json
+          gon -log-level=debug -log-json ./gon-arm64.json
+      - name: rename signed binaries
+        run: |
+          mv release/signed/kbc-amd64.zip release/signed/kbc_${{ needs.set_version.outputs.version }}_darwin_amd64.zip
+          mv release/signed/kbc-arm64.zip release/signed/kbc_${{ needs.set_version.outputs.version }}_darwin_arm64.zip
```
Comment (kacurez) [inline .github/workflows/push.yml:185]: po podpisani premenujem zip subry na povodne hodnoty tj `kbc_<version>_darwin_<arch>.zip`

File: .github/workflows/push.yml [hunk 4]
```diff
@@ -134,3 +137,57 @@ jobs:
...
+      - name: Upload binaries to release
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: release/signed/*
+          tag: ${{ github.ref }}
+          overwrite: true
+          file_glob: true
```
Comment (kacurez) [inline .github/workflows/push.yml:193]: zkopirovana povodna github action(pouzivana pred goreleaserom) na release signed zipov

File: .goreleaser.yml [hunk 5]
```diff
@@ -16,13 +17,31 @@ builds:
...
+    env:
+      - CGO_ENABLED=0
+    ldflags:
+      - -s -w
+      - -X keboola-as-code/src/build.BuildVersion={{ if index .Env "TARGET_VERSION"  }}{{ .Env.TARGET_VERSION }}{{ else }}dev{{ end }}
+      - -X keboola-as-code/src/build.BuildDate={{.Date}}
+      - -X keboola-as-code/src/build.GitCommit={{.Commit}}
+    hooks:
+      # Copy binary to TARGET_PATH if set
+      post: '{{ if index .Env "TARGET_PATH"  }}cp "{{ .Path }}" "{{ .Env.TARGET_PATH }}"{{ else }}true{{ end }}'
```
Comment (kacurez) [inline .goreleaser.yml:35]: separovany build pre darwin aby nebuildoval zip archivy.

File: .github/workflows/push.yml [hunk 6]
```diff
@@ -134,3 +137,57 @@ jobs:
         env:
           TARGET_VERSION: ${{ needs.set_version.outputs.version }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  release-macos:
```
Comment (webrouse) [inline .github/workflows/push.yml:140]: Mohli by sme to prosim nazvat `release-mac-os`, pre lepsiu citatelnost?

File: .github/workflows/push.yml [hunk 7]
```diff
@@ -134,3 +137,57 @@ jobs:
...
+          brew tap mitchellh/gon
+          brew install mitchellh/gon/gon
+          brew install tree
+          tree release
+      - name: Sign the mac binaries with Gon
+        env:
+          AC_USERNAME: ${{ secrets.AC_USERNAME }}
+          AC_PASSWORD: ${{ secrets.AC_PASSWORD }}
+        run: |
+          gon -log-level=debug -log-json ./gon-amd64.json
```
Comment (webrouse) [inline .github/workflows/push.yml:180]: Je potrebny JSON log? Neda sa tam z toho potom velmi nic vycitat v CI.

File: gon-amd64.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
+{
+    "source": [
+        "./release/darwin-amd64/kbc"
+    ],
+    "bundle_id": "com.keboola.kbc",
+    "apple_id": {
+        "password": "@env:AC_PASSWORD"
+    },
+    "sign": {
+        "application_identity": "Developer ID Application: Tomas Netrval"
```
Comment (webrouse) [inline gon-amd64.json:10]: Neda sa tu dat `Keboola`?

Comment (kacurez) [reply gon-amd64.json:10]: imo neda, musel by sa vygenerovat novy certifikat
image

File: .github/workflows/push.yml (hunk 7 above)
Comment (kacurez) [reply .github/workflows/push.yml:180]: zmenim log level na `info`

File: gon-amd64.json (hunk 8 above)
Comment (webrouse) [reply gon-amd64.json:10]: ok

File: .github/workflows/push.yml (hunk 7 above)
Comment (kacurez) [reply .github/workflows/push.yml:180]: vid https://github.com/mitchellh/gon#machine-readable-output pise `Note you must specify both -log-level and -log-json.` - dam to aspon na to info.

**Review by webrouse — APPROVED** [review]

**Pull request merged**